### ISSUE TITLE
2.0 and removing deprecated features?

### ISSUE THREAD

**Issue by truckingsim** [truckingsim/Ajax-Bootstrap-Select#47]
I think we work on ripping out the hackiness that we have because we didn't have events.
I think we have also deprecated the old features long enough.

* So bump up the min version of bootstrap-select
* Rip out hacky code for starting up bootstrap-select
* Rip out code that deals with deprecation.

**Comment by markcarver** [comment]
>* So bump up the min version of bootstrap-select to the version that added events
>* Rip out hacky code for starting up bootstrap-select

Yes, but I'm not entirely sure an `initialize` event has actually been added yet, see https://github.com/silviomoreto/bootstrap-select/issues/699

>* Rip out code that deals with deprecation

I'm fine with that. We can always add it back in from a previous commit if needed.

I'm not entirely sure a 2.0.0 version should just be "ripping stuff out" though. At the very least we should probably attempt to add two features that have been sitting in the issue queue for a bit, #31 and #32. I'll add the milestone to those two issues at least.

**Comment by markcarver** [comment]
I've created #48 for the `initialize` event issue.

**Comment by truckingsim** [comment]
Agreed on #31 and #32.  It also looks like Casey is at least being responsive about the init event so hopefully that will come around in time.

I'll start looking into #32 and see where I come to on that about whether its even feasible to do in any kind of reasonable way.  I also think we should make another round on the current open bugs, look into whether any of them are anything we can do about, and also looking if maybe something that wasn't really fixable before maybe now (i.e. title stuff) with updated version of bootstrap-select.

**Comment by markcarver** [comment]
As a side note, it'd also be nice to convert all this to ES6

**Comment by truckingsim** [comment]
@markcarver for sure, would be very nice.

**Comment by AlokArora** [comment]
How about highlighting the search text in options as they show up filtered from server side? or even from localStorage.
